File: kt/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/ExpressionStatementSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class ExpressionStatementSyntax(val expression: ExpressionSyntax) : StatementSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.ExpressionStatement
    override val children: List<SyntaxNode>
        get() = listOf(expression)
}

File: cs/Minsk/CodeAnalysis/Syntax/ExpressionStatementSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public sealed class ExpressionStatementSyntax : StatementSyntax
{
    public ExpressionSyntax Expression { get; }

    public ExpressionStatementSyntax(ExpressionSyntax expression)
    {
        Expression = expression;
    }

    public override SyntaxKind Kind => SyntaxKind.ExpressionStatement;
    public override IEnumerable<SyntaxNode> Children => new[] { Expression };
}
